zerver/lib/user_groups.py

```diff
@@ -411,73 +413,121 @@ def set_defaults_for_group_settings(
         else:
             default_group_name = permission_config.default_group_name
 
-        default_group = system_groups_name_dict[default_group_name]
+        default_group = system_groups_name_dict[default_group_name].usergroup_ptr
         setattr(user_group, setting_name, default_group)
 
+        setting_name_for_named_object = "named_group_" + setting_name
+        setattr(user_group, setting_name_for_named_object, default_group)
+
     return user_group
 
 
+def bulk_create_system_user_groups(groups: List[Dict[str, str]], realm: Realm) -> None:
+    # This value will be used to set the temporary initial value for different
+    # settings since we can only set them to the correct values after the groups
+    # are created.
+    initial_group_setting_value = -1
+    rows = [
+        SQL("({},{},{},{},{})").format(
+            Literal(realm.id),
+            Literal(group["name"]),
+            Literal(group["description"]),
+            Literal(True),
+            Literal(initial_group_setting_value),
+        )
+        for group in groups
+    ]
+
+    query = SQL(
+        """
+        INSERT INTO zerver_usergroup (realm_id, name, description, is_system_group, can_mention_group_id)
+        VALUES {rows}
+        RETURNING id
+        """
+    ).format(rows=SQL(", ").join(rows))
+    with connection.cursor() as cursor:
+        cursor.execute(query)
+        user_group_ids = [id for (id,) in cursor.fetchall()]
+
+    rows = [
+        SQL("({},{},{},{},{},{})").format(
+            Literal(user_group_ids[idx]),
+            Literal(realm.id),
+            Literal(group["name"]),
+            Literal(group["description"]),
+            Literal(True),
+            Literal(initial_group_setting_value),
+        )
+        for idx, group in enumerate(groups)
+    ]
+    query = SQL(
+        """
+        INSERT INTO zerver_namedusergroup (usergroup_ptr_id, realm_id, name, description, is_system_group, can_mention_group_id)
+        VALUES {rows}
```

Comment:
Have used two separate queries for now - one for `UserGroup` and one for `NamedUserGroup` objects. Tried doing it in one query, but wasn't able to do it. Any suggestions would be helpful.

Reply:
Seems like a fine thing to post in #backend for advice, but not a blocker for this PR; it's all in one transaction.